pkg/loopback/loop_module.go

```diff
@@ -0,0 +1,177 @@
+// +build linux,cgo
+
+package loopback // import "github.com/docker/docker/pkg/loopback"
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"math/bits"
+	"os"
+	"strconv"
+	"syscall"
+
+	"github.com/docker/docker/pkg/system"
+)
+
+const (
+	loopMajorDev = 7 // see /usr/include/linux/major.h
+	loopControl = "/dev/loop-control"
+	loopFormat = "/dev/loop%d"
+	loopZero = "/dev/loop0"
+	sysfsModuleFormat = "/sys/module/loop/parameters/%s"
+)
+
+type loopModuleContext interface {
+	PerformPathStat(path string) (os.FileInfo, error)
+	GetNextFreeDeviceIndex() (int, error)
+	GetBaseDeviceNodeStat() (*syscall.Stat_t, error)
+	OpenSysfsParameterFile(param string) (io.ReadCloser, error)
+	GetMaxPartitionParameter() (uint, error)
+	GetPartitionShift() (uint, error)
+	GetMknodDeviceNumber(index int) (int, error)
+	PerformMknod(path string, mode uint32, dev int) error
+	MakeIndexNode(index int) (os.FileInfo, error)
+}
+
```

Comment:
Are you introducing this interface in order to be able to test the package? Problem is, interfaces are slow, so I would not use it unless we're going to abstract various loopback types (we don't).